spec/std/number_spec.cr

```diff
@@ -299,6 +299,24 @@ describe "Number" do
         end
       end
     end
+
+    describe "whole range" do
+        it { (UInt8::MIN..UInt8::MAX).each.count { true }.should eq(256) }
+        it_iterates "UInt8 upwards", (UInt8::MIN.to_i..UInt8::MAX.to_i).map(&.to_u8), (UInt8::MIN..UInt8::MAX).step(by: 1)
+        it_iterates "UInt8 downwards", (UInt8::MIN.to_i..UInt8::MAX.to_i).map(&.to_u8).reverse, (UInt8::MAX..UInt8::MIN).step(by: -1)
+
+        it { (Int8::MIN..Int8::MAX).each.count { true }.should eq(256) }
+        it_iterates "Int8 upwards", (Int8::MIN.to_i..Int8::MAX.to_i).map(&.to_i8), (Int8::MIN..Int8::MAX).step(by: 1)
+        it_iterates "Int8 downwards", (Int8::MIN.to_i..Int8::MAX.to_i).map(&.to_i8).reverse, (Int8::MAX..Int8::MIN).step(by: -1)
+
+        it { (Int16::MIN..Int16::MAX).each.count { true }.should eq(65536) }
+        it_iterates "Int16 upwards", (Int16::MIN.to_i..Int16::MAX.to_i).map(&.to_i16), (Int16::MIN..Int16::MAX).step(by: 1)
+        it_iterates "Int16 downwards", (Int16::MIN.to_i..Int16::MAX.to_i).map(&.to_i16).reverse, (Int16::MAX..Int16::MIN).step(by: -1)
```

Comment:
Collecting the entire ranges in arrays isn't super efficient, but not a big deal either. Using `it_iterates` ensures these test are run against both `step` methods.